java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -272,19 +312,22 @@
  */
 JNIEXPORT jshort JNICALL Java_ai_onnxruntime_OnnxTensor_getShort
   (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle, jint onnxType) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
+    (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
     const OrtApi* api = (const OrtApi*) apiHandle;
   if (onnxType == 3) {
```

Comment:
>if (onnxType == 3) { [](http://example.com/codeflow?start=2&length=20)

Same comments as above.
enum + combine the code.

Reply:
Fixed.

Reply:
My opinion on that is that people who get unsigned values back in Java should deal with that themselves. The bit pattern is correct, and if you're working in unsigned types you should understand how to sort that out for your use case. 

There is no bigger primitive type than long, so we don't have a good solution for that one because you'd end up making an array of objects to hold it and it quickly becomes intractable. When Java gets value types then we'll be able to define unsigned types and return those, but for the moment people usually expect to get the signed versions and if they need something else it's on them to figure it out from the bits.